core/src/main/java/tachyon/web/UIWebServer.java

```diff
@@ -57,10 +59,16 @@
   public UIWebServer(String serverName, InetSocketAddress address, MasterInfo masterInfo) {
     mAddress = address;
     mServerName = serverName;
-    mServer = new Server(mAddress);
+    mServer = new Server();
+    SelectChannelConnector connector = new SelectChannelConnector();
+    connector.setPort(address.getPort());
+    connector.setAcceptors(MasterConf.get().WEB_THREAD_COUNT);
+    mServer.setConnectors(new Connector[] { connector });
 
     QueuedThreadPool threadPool = new QueuedThreadPool();
-    threadPool.setMaxThreads(MasterConf.get().WEB_THREAD_COUNT);
+    // Jetty needs at least (1 + selectors + acceptors) threads.
+    threadPool.setMinThreads(MasterConf.get().WEB_THREAD_COUNT * 2 + 1);
+    threadPool.setMaxThreads(MasterConf.get().WEB_THREAD_COUNT * 2 + 100);
```

Comment:
MaxThreads = MinThreads means that there will be only one worker thread doing the actual jobs. If multiple clients access the web UI simultaneously, Jetty server creates or gets several work threads from the thread pool to process the jobs. The thread pool also thrinks its size when there are too many idle worker threads until it hits the MinThreads. The MaxThreads is a just a limit on the thread pool, by default it is 254. When the web server is idle, it only maintains MinThreads threads.